### PR TITLE
etcd-runner: clean up watcher runner

### PR DESCRIPTION
**Pull request by xiang90**
/cc @sinsharat @heyitsanthony

### CODE REVIEW

File: tools/functional-tester/etcd-runner/watcher.go [hunk 1]
```diff
@@ -149,18 +115,56 @@ func performWatchOnPrefixes(ctx context.Context, getClient getClientFunc, round
 		rc.Close()
 	}
 
-	deletePrefixWithRety(client, ctx, roundPrefix)
+	deletePrefix(ctx, client, roundPrefix)
+}
+
+func checkWatchResponse(ctx context.Context, wc clientv3.WatchChan, prefix string, keys []string) {
```
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/watcher.go]: `ctx` already bound to `wc`, so no need to pass around?

File: tools/functional-tester/etcd-runner/watcher.go [hunk 2]
```diff
@@ -149,18 +115,56 @@ func performWatchOnPrefixes(ctx context.Context, getClient getClientFunc, round
...
+				}
+				n++
+			}
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func putKey(ctx context.Context, client *clientv3.Client, key string) error {
```
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/watcher.go]: `putKeyOnce`?

Comment (xiang90) [inline tools/functional-tester/etcd-runner/watcher.go]: Why once? i think we do retry here at line 146.

**Comment by xiang90** [comment]
@heyitsanthony Fixed. PTAL.

File: tools/functional-tester/etcd-runner/watcher.go (hunk 2 above)
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/watcher.go]: but it's guarded by the txn so it'll only bump the revision by at most 1-- maybe `putKeyAtMostOnce`?

Comment (xiang90) [inline tools/functional-tester/etcd-runner/watcher.go]: Oh... Now I understand. Will make this clear.

File: tools/functional-tester/etcd-runner/watcher.go [hunk 3]
```diff
@@ -149,18 +115,54 @@ func performWatchOnPrefixes(ctx context.Context, getClient getClientFunc, round
...
 	}
 
-	deletePrefixWithRety(client, ctx, roundPrefix)
+	deletePrefix(ctx, client, roundPrefix)
+}
+
+func checkWatchResponse(ctx context.Context, wc clientv3.WatchChan, prefix string, keys []string) {
+	for n := 0; n < len(keys); {
+		select {
+		case wr := <-wc:
```
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/watcher.go]: `case wr, ok := <- wc:` so it'll pick up the cancelation?

**Comment by sinsharat** [comment]
LGTM with @heyitsanthony suggestions implemented.
Thank you!

**Comment by xiang90** [comment]
@heyitsanthony All fixed. PTAL.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**